application/models/SurveysGroups.php

```diff
@@ -339,17 +390,95 @@ public static function model($className = __CLASS__)
     protected static function getPermissionCriteria()
     {
         $criteriaPerm = new CDbCriteria;
-        if (!Permission::model()->hasGlobalPermission("surveys", 'read')) {
+        if (!Permission::model()->hasGlobalPermission("surveys", 'read') || !Permission::model()->hasGlobalPermission("surveysgroups", 'read')) {
+            /* owner of surveygroup */
+            $criteriaPerm->compare('t.owner_id', Yii::app()->user->id, false);
+            /* Simple permission on SurveysGroup inside a group */
+            $criteriaPerm->mergeWith(array(
+                'join'=>"LEFT JOIN {{permissions}} AS permissions ON (permissions.entity_id = t.gsid AND permissions.permission='group' AND permissions.entity='surveysgroups' AND permissions.uid='".Yii::app()->user->id."') ",
+            ));
+            $criteriaPerm->compare('permissions.read_p', '1', false, 'OR');
+            /* Permission on Survey inside a group */
             $criteriaPerm->mergeWith(array(
                 'join'=>"LEFT JOIN {{surveys}} AS surveys ON (surveys.gsid = t.gsid)
-                        LEFT JOIN {{permissions}} AS permissions ON (permissions.entity_id = surveys.sid AND permissions.permission='survey' AND permissions.entity='survey' AND permissions.uid='".Yii::app()->user->id."') ",
+                        LEFT JOIN {{permissions}} AS surveypermissions ON (surveypermissions.entity_id = surveys.sid AND surveypermissions.permission='survey' AND surveypermissions.entity='survey' AND surveypermissions.uid='".Yii::app()->user->id."') ",
             ));
-            $criteriaPerm->compare('t.owner_id', Yii::app()->user->id, false);
             $criteriaPerm->compare('surveys.owner_id', Yii::app()->user->id, false, 'OR');
-            $criteriaPerm->compare('permissions.read_p', '1', false, 'OR');
-            $criteriaPerm->compare('t.gsid', '1', false, 'OR');  // "default" survey group
+            $criteriaPerm->compare('surveypermissions.read_p', '1', false, 'OR');
+            /* default survey group is always avaliable */
+            $criteriaPerm->compare('t.gsid', '1', false, 'OR');
+            /* survey group set as avaiable */
+            $criteriaPerm->compare('t.alwaysavailable', '1', false, 'OR'); // Is public
         }
         return $criteriaPerm;
     }
 
+    /**
+     * Get Permission data for Permission object
+     * @param string $key
+     * @return array
```

Comment:
This is why data-value objects are better than arrays. ^^ `array` doesn't tell us anything about the structure of the data.

Reply:
It's this function : https://github.com/LimeSurvey/LimeSurvey/blob/ae7b04859574870146df20696bfa41807b4c1663/application/models/Permission.php#L66 moved to Object … i think it's must be done

Reply:
Yep